src/component/component_system_manager.cc

```diff
@@ -14,6 +14,19 @@ extern frame_info frame_info;
 
 extern particle_manager *particle_man;
 
+static bool
+filter_matches_message(comms_msg const &msg, wire_filter_ptr const &filter) {
+    auto &cwire_man = component_system_man.managers.wire_comms_component_man;
+    auto sender = cwire_man.get_instance_data(msg.originator);
+
+    /* If we have no filter defined, we match everything. */
+    if (!filter.wrapped)
+        return true;
+
+    /* We have a filter. Sender must have a filter, and it must match. */
```

Comment:
sender must have a label?

Reply:
If filter requires a label, then yes.

Reply:
The first case  above catches the case where sender doesnt have a label and filter doesnt require anything

Reply:
OH -- are you complaining about my slightly wrong comment, rather than the logic?

Reply:
The comment says filter. But sender doesn't require a filter and you check for a label.

Reply:
Yes